plugins/woocommerce/tests/legacy/unit-tests/order/class-wc-tests-order-functions.php

```diff
@@ -178,11 +184,30 @@ public function test_wc_get_order() {
 		$post = $this->factory->post->create_and_get( array( 'post_type' => 'post' ) );
 		$this->assertFalse( wc_get_order( $post->ID ) );
 
+		// Assert the return when $the_order args is a random (incorrect) id.
+		$this->assertFalse( wc_get_order( 123456 ) );
+
 		// Assert the return when $the_order args is false.
 		$this->assertFalse( wc_get_order( false ) );
 
-		// Assert the return when $the_order args is a random (incorrect) id.
-		$this->assertFalse( wc_get_order( 123456 ) );
+		$post = get_post( $order->get_id() );
+		$this->assertInstanceOf(
+			'WC_Order',
+			wc_get_order(),
+			'If no order ID is specified, wc_get_order() will use the global $post object to try and determine the current order.'
+		);
+
+		unset( $post );
+		$theorder = $order;
+		$this->assertInstanceOf(
+			'WC_Order',
+			wc_get_order(),
+			'If no order ID is specified, wc_get_order() will use the global $theorder object to try and determine the current order.'
```

Comment:
I think you mean "If the global $post object is not defined..." here?

Reply:
Hmm 🤔 

That's actually not what I meant, and even if I didn't unset `$post` the assertion should pass, but you asking this makes me think I haven't laid things out very clearly.

`wc_get_order()` will potentially make use of the global `$theorder` object ***or*** the global `$post` object (and, in fact, the former takes precedence over the latter, though I decided not to highlight that subtlety here). The only reason I unset `$post` was to remove it from the landscape and show, as the description notes, that the SUT will also look to `$theorder` to figure out the current order.

Is there a way I could make this clearer?